compiler/natives/src/internal/reflectlite/reflectlite.go

```diff
@@ -0,0 +1,973 @@
+			rt.tflag |= tflagUncommon
+			if typ.Get("named").Bool() {
+				rt.tflag |= tflagNamed
+			}
+			var reflectMethods []method
+			for i := 0; i < methodSet.Length(); i++ { // Exported methods first.
+				m := methodSet.Index(i)
+				exported := internalStr(m.Get("pkg")) == ""
+				if !exported {
+					continue
+				}
+				reflectMethods = append(reflectMethods, method{
+					name: newNameOff(newName(internalStr(m.Get("name")), "", exported)),
+					mtyp: newTypeOff(reflectType(m.Get("typ"))),
+				})
+			}
+			xcount := uint16(len(reflectMethods))
+			for i := 0; i < methodSet.Length(); i++ { // Unexported methods second.
+				m := methodSet.Index(i)
+				exported := internalStr(m.Get("pkg")) == ""
+				if exported {
+					continue
+				}
+				reflectMethods = append(reflectMethods, method{
+					name: newNameOff(newName(internalStr(m.Get("name")), "", exported)),
+					mtyp: newTypeOff(reflectType(m.Get("typ"))),
+				})
+			}
+			ut := &uncommonType{
+				pkgPath:  newNameOff(newName(internalStr(typ.Get("pkg")), "", false)),
+				mcount:   uint16(methodSet.Length()),
+				xcount:   xcount,
+				_methods: reflectMethods,
+			}
+			uncommonTypeMap[rt] = ut
+			js.InternalObject(ut).Set(idJsType, typ)
+		}
+
+		switch rt.Kind() {
+		case Array:
+			setKindType(rt, &arrayType{
+				elem: reflectType(typ.Get("elem")),
+				len:  uintptr(typ.Get("len").Int()),
+			})
+		case Chan:
+			dir := BothDir
+			if typ.Get("sendOnly").Bool() {
+				dir = SendDir
+			}
+			if typ.Get("recvOnly").Bool() {
+				dir = RecvDir
+			}
+			setKindType(rt, &chanType{
+				elem: reflectType(typ.Get("elem")),
+				dir:  uintptr(dir),
+			})
+		case Func:
+			params := typ.Get("params")
+			in := make([]*rtype, params.Length())
+			for i := range in {
+				in[i] = reflectType(params.Index(i))
+			}
+			results := typ.Get("results")
+			out := make([]*rtype, results.Length())
+			for i := range out {
+				out[i] = reflectType(results.Index(i))
+			}
+			outCount := uint16(results.Length())
+			if typ.Get("variadic").Bool() {
+				outCount |= 1 << 15
+			}
+			setKindType(rt, &funcType{
+				rtype:    *rt,
+				inCount:  uint16(params.Length()),
+				outCount: outCount,
+				_in:      in,
+				_out:     out,
+			})
+		case Interface:
+			methods := typ.Get("methods")
+			imethods := make([]imethod, methods.Length())
+			for i := range imethods {
+				m := methods.Index(i)
+				imethods[i] = imethod{
+					name: newNameOff(newName(internalStr(m.Get("name")), "", internalStr(m.Get("pkg")) == "")),
+					typ:  newTypeOff(reflectType(m.Get("typ"))),
+				}
+			}
+			setKindType(rt, &interfaceType{
+				rtype:   *rt,
+				pkgPath: newName(internalStr(typ.Get("pkg")), "", false),
+				methods: imethods,
+			})
+		case Map:
+			setKindType(rt, &mapType{
+				key:  reflectType(typ.Get("key")),
+				elem: reflectType(typ.Get("elem")),
+			})
+		case Ptr:
+			// p := (*ptrType)(unsafe.Pointer(rt))
+			// p.elem = reflectType(typ.Get("elem"))
```

Comment:
Dead code? Or TODO?

Reply:
Yep, it's a dead code I missed when adopting changes from @visualfc's fork. Removed.